test/cluster/cluster_test.go

```diff
@@ -767,6 +772,74 @@ func ScaleDownWithMigrateFillDelay(ctx goctx.Context) {
 	)
 }
 
+func ValidateScaleDownWaitForMigrations(ctx goctx.Context) {
+	Context(
+		"ValidateScaleDownWaitForMigrations", func() {
+			clusterNamespacedName := test.GetNamespacedName(
+				"wait-for-migrations-cluster", namespace,
+			)
+
+			BeforeEach(
+				func() {
+					aeroCluster := createDummyAerospikeCluster(clusterNamespacedName, 4)
+					Expect(DeployCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+				},
+			)
+
+			AfterEach(
+				func() {
+					aeroCluster := &asdbv1.AerospikeCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      clusterNamespacedName.Name,
+							Namespace: clusterNamespacedName.Namespace,
+						},
+					}
+
+					Expect(DeleteCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+					Expect(CleanupPVC(k8sClient, aeroCluster.Namespace, aeroCluster.Name)).ToNot(HaveOccurred())
+				},
+			)
+
+			It(
+				"Should wait for migrations to complete before pod deletion", func() {
+					aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+					Expect(err).ToNot(HaveOccurred())
+
+					aeroCluster.Spec.Size -= 1
+					err = k8sClient.Update(ctx, aeroCluster)
+					Expect(err).ToNot(HaveOccurred())
+
+					Eventually(func() bool {
+						podList, lErr := getPodList(aeroCluster, k8sClient)
+						Expect(lErr).ToNot(HaveOccurred())
+						migrations, mErr := getMigrationsInProgress(ctx, k8sClient, clusterNamespacedName)
+						Expect(mErr).ToNot(HaveOccurred())
+
+						// Fail condition: pods scaledown while migrations still running
+						if utils.Len32(podList.Items) < 4 && migrations != 0 {
+							Fail(fmt.Sprintf("Pods scaledown while migrations still running (pods=%d, migrations=%d)",
+								utils.Len32(podList.Items), migrations))
+						}
+
+						// Success condition: migrations complete and number of pods is equal to spec size
+						if utils.Len32(podList.Items) == aeroCluster.Spec.Size && migrations == 0 {
+							return true
+						}
+						// Keep waiting otherwise
+						return false
+					}, 5*time.Minute, 10*time.Second).Should(BeTrue(), "Pods should only scaledown after migrations complete")
```

Comment:
Are we sure that this 10s interval is sufficient to check the required condition?
Seems like here there may be race conditions here. Is it possible to check the expected behaviour like 4 nodes + 1 quiesced  + migration non-zeo? This is confirm the behaviour as per expected change